### ISSUE TITLE
bug: no-construct-in-public-property-of-construct rule doesn't detect generic types

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#222]
### Description
The `no-construct-in-public-property-of-construct` rule fails to detect CDK Construct types when they are wrapped in generic types (e.g. `Readonly<T>`).

### Example Code
```typescript
import { Construct } from 'constructs';
import { Bucket } from 'aws-cdk-lib/aws-s3';

export class MyConstruct extends Construct {
  // These should be detected as errors but are not
  public readonly bucket1: Readonly<Bucket>;
  public readonly bucket2: Partial<Bucket>;
  public readonly buckets: Array<Bucket>;
  
  constructor(scope: Construct, id: string) {
    super(scope, id);
    this.bucket1 = new Bucket(this, 'Bucket1');
    this.bucket2 = new Bucket(this, 'Bucket2');
    this.buckets = [new Bucket(this, 'Bucket3')];
  }
}
```

### Expected Behavior
The rule should detect and report errors for all public properties that contain CDK Construct types, even when wrapped in generic types like:
- `Readonly<Bucket>`
- `Partial<Bucket>`
- `Array<Bucket>`
- Custom generic wrappers

### Actual Behavior
The rule only detects direct Construct type usage and array syntax (`Bucket[]`), but misses generic type wrappers.

### Environment
- eslint-cdk-plugin Version: 3.4.0
- Node.js Version: 22.15.0
- Language: TypeScript

### Related Issue
This is the same issue as #220 but affecting the `no-construct-in-public-property-of-construct` rule instead of `no-construct-in-interface`.
